### PR TITLE
fix(core): remove hammerjs from library

### LINKED ISSUE #4132: BUG: dependency on CommonJS modules
Warning on ng serve

When I start the angular application with the fundamental-ngx  I have:

Warning: C:\git\x-kafka-cockpit-ui\node_modules\@fundamental-ngx\core\__ivy_ngcc__\fesm2015\fundamental-ngx-core.js depends on 'hammerjs'. CommonJS or AMD dependencies can cause optimization bailouts.       
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Angular version: 10.2.1
Fundamental-NGX version: 0.24.2

this is only a warning, but it is recommended to not have dependencies on CommonJS modules.

Thanks in advance
Best Regards,

Silvia Giordano

### PR DESCRIPTION
**Pull request by N1XUS**
#### Please provide a link to the associated issue.
Closes #4132 

#### Please provide a brief summary of this pull request.
Removed hammerjs library from fundamental-ngx/core and refactored carousel component
#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [x] Run npm run build-pack-library and test in external application

Documentation checklist:
- n/a update `README.md`
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

File: libs/core/src/lib/carousel/carousel.component.scss [hunk 1]
```diff
@@ -3,3 +3,16 @@
...
+        touch-action: pan-x;
+        user-select: none;
+        -webkit-user-drag: none;
+        -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
+
+        &.fd-carousel__content--horizontal {
+            touch-action: pan-y;
+        }
+    }
+}
```
Comment (InnaAtanasova) [inline libs/core/src/lib/carousel/carousel.component.scss:18]: Please move this to fundamental-styles

File: libs/core/src/lib/carousel/carousel.service.ts [hunk 2]
```diff
@@ -76,13 +74,27 @@ export class CarouselService implements OnDestroy {
...
+    /** @hidden */
+    private _initialDragPosition = 0;
+    /** @hidden */
+    private _lastDragPosition = 0;
+    /** @hidden */
+    private _listenToMouseMove = false;
+    /** @hidden */
+    private _dragStarted = false;
+
+    /** An RxJS Subject that will kill the data stream upon component’s destruction (for unsubscribing)  */
```
Comment (InnaAtanasova) [inline libs/core/src/lib/carousel/carousel.service.ts:88]: Should be hidden

File: libs/core/src/lib/carousel/carousel.service.ts [hunk 3]
```diff
@@ -319,19 +331,70 @@ export class CarouselService implements OnDestroy {
...
-            this._hammer.get('pan').set({ direction: Hammer.DIRECTION_VERTICAL });
-            this._hammer.on('panmove', (event) => this._handlePan(event.deltaY));
-            this._hammer.on('panstart', () => this._handlePanStart());
-            this._hammer.on('panend', (event) => this._handlePanEnd(event.deltaY));
-        } else {
-            this._hammer.on('panmove', (event) => this._handlePan(event.deltaX));
-            this._hammer.on('panstart', () => this._handlePanStart());
-            this._hammer.on('panend', (event) => this._handlePanEnd(event.deltaX));
-        }
+    private _subscribeToEvents(
```
Comment (InnaAtanasova) [inline libs/core/src/lib/carousel/carousel.service.ts:340]: /** @hidden */

**Review by InnaAtanasova — COMMENTED** [review]
minor stuff, the code looks great

File: libs/core/src/lib/carousel/carousel.component.scss (hunk 1 above)
Comment (N1XUS) [reply libs/core/src/lib/carousel/carousel.component.scss:19]: I've created separate bug in fundamental-styles to reflect the carousel component's layout from fundamental-ngx: https://github.com/SAP/fundamental-styles/issues/2574

**Review by InnaAtanasova — APPROVED** [review]

File: libs/core/src/lib/package.json [hunk 4]
```diff
@@ -25,7 +25,6 @@
     "@sap-theming/theming-base-content": "THEMING_VER_PLACEHOLDER",
     "focus-trap": "FOCUSTRAP_VER_PLACEHOLDER",
     "fundamental-styles": "FDSTYLES_VER_PLACEHOLDER",
-    "hammerjs": "HAMMERJS_VER_PLACEHOLDER",
```
Comment (kavya-b) [inline libs/core/src/lib/package.json:28]: maybe we can remove the placeholder from `sync-versions.sh` too?

File: ci-scripts/sync-version.sh [hunk 5]
```diff
@@ -25,7 +24,5 @@ grep -rl 'RXJS_VER_PLACEHOLDER' . | xargs  perl -X -p -i -e "s/RXJS_VER_PLACEHOL
 grep -rl 'CDK_VER_PLACEHOLDER' . | xargs  perl -X -p -i -e "s/CDK_VER_PLACEHOLDER/${CDK_VERSION}/g"
 grep -rl 'FDSTYLES_VER_PLACEHOLDER' . | xargs  perl -X -p -i -e "s/FDSTYLES_VER_PLACEHOLDER/${FDSTYLES_VERSION}/g"
 grep -rl 'FOCUSTRAP_VER_PLACEHOLDER' . | xargs  perl -X -p -i -e "s/FOCUSTRAP_VER_PLACEHOLDER/${FOCUSTRAP_VERSION}/g"
-grep -rl 'THEMING_VER_PLACEHOLDER' . | xargs  perl -X -p -i -e "s/THEMING_VER_PLACEHOLDER/${THEMING_VERSION}/g"
```
Comment (kavya-b) [inline ci-scripts/sync-version.sh:28]: we do need `THEMING_VER_PLACEHOLDER` right? I don't think this should be removed.

**Review by Betrozov — APPROVED** [review]

**Pull request merged**